llvm/lib/Target/AIE/AIE2Schedule.td

```diff
@@ -448,6 +455,13 @@ InstrItinData<II_MOVXM, [PrefixCycle<P_WM_PORT>, PrefixCycle<S_WM_PORT>, PrefixC
 InstrItinData<II_MOV_SCL, [PrefixCycle<P_RM_PORT>, PrefixCycle<R_WM_PORT>, PrefixCycle<P_WM_PORT>, PrefixCycle<M_WM_PORT>,
                            PrefixCycle<DJ_WM_PORT>, PrefixCycle<DN_WM_PORT>, PrefixCycle<DC_WM_PORT>, SimpleCycle<S_WM_PORT>],
                            [1,1]>,
+InstrItinData<II_MOV_SCL_DC, [PrefixCycle<P_RM_PORT>, SimpleCycle<DC_WM_PORT>], [1,1]>,
+InstrItinData<II_MOV_SCL_DJ, [PrefixCycle<P_RM_PORT>, SimpleCycle<DJ_WM_PORT>], [1,1]>,
+InstrItinData<II_MOV_SCL_DN, [PrefixCycle<P_RM_PORT>, SimpleCycle<DN_WM_PORT>], [1,1]>,
+InstrItinData<II_MOV_SCL_M, [PrefixCycle<P_RM_PORT>, SimpleCycle<M_WM_PORT>], [1,1]>,
+InstrItinData<II_MOV_SCL_P, [PrefixCycle<P_RM_PORT>, SimpleCycle<P_WM_PORT>], [1,1]>,
+InstrItinData<II_MOV_SCL_R, [PrefixCycle<P_RM_PORT>, SimpleCycle<R_WM_PORT>], [1,1]>,
+InstrItinData<II_MOV_SCL_S, [PrefixCycle<P_RM_PORT>, SimpleCycle<S_WM_PORT>], [1,1]>,
```

Comment:
So we have a different port for shift registers?

Reply:
That is not a case for slot "M", writing to "R" / "S" share the same port.
Have added a commit first fixing that followed by all my changes.